twitter/Models/Tweet.m

```diff
@@ -0,0 +1,65 @@
+//
+//  Tweet.m
+//  twitter
+//
+//  Created by Joey R. Hernandez Perez on 6/29/20.
+//  Copyright © 2020 Emerson Malca. All rights reserved.
+//
+
+#import "Tweet.h"
+#import "User.h"
+
+@implementation Tweet
+
+- (instancetype) initWithDictionary:(NSDictionary *) dictionary {
+    
+    self = [super init];
+    
+    if (self) {
+        
+        // Is this a re-tweet?
+        NSDictionary *originalTweet = dictionary[@"retweeted_status"];
+        if(originalTweet != nil){
+            NSDictionary *userDictionary = dictionary[@"user"];
+            self.retweetedByUser = [[User alloc] initWithDictionary:userDictionary];
+
+            // Change tweet to original tweet
+            dictionary = originalTweet;
+        }
+        self.idStr = dictionary[@"id_str"];
```

Comment:
I would add type checks on literally all of these. 

Try to add as type-checking and nil checking helper since it will become very verbose.

If you get stuck there's examples here:
https://github.com/facebook/facebook-ios-sdk/blob/e01fcbbe1e3d54cac15a141fe5f6cc0d16bec35f/FBSDKCoreKit/FBSDKCoreKit/Basics/Internal/FBSDKTypeUtility.h